src/ethereum/byzantium/vm/error.py

```diff
@@ -12,24 +12,43 @@
 Errors which cause the EVM to halt exceptionally.
 """
 
+from ethereum.exceptions import EthereumException
 
-class StackUnderflowError(Exception):
+
+class ConsumeAllGasException(EthereumException):
```

Comment:
I don't believe there is standard terminology for this. The Yellow Paper calls this "Exceptional Halt" while most EIPs use "Out Of Gas Error". "Consumes all gas" is also common in prose.

I went with `ConsumeAllGasException`, because I want to maximize clarity for the uninformed reader who will be confused by `OutOfGasError` and have to look up the meaning of `ExceptionalHalt`.

Reply:
I think the name of the class should represent the cause of the exception (aka what it _is_), and the docstring can describe its effects. This is more consistent with `InvalidBlock` and the other VM errors.